src/rpp/rpp/operators/buffer.hpp

```diff
@@ -0,0 +1,83 @@
+//                  ReactivePlusPlus library
+//
+//          Copyright Aleksey Loginov 2022 - present.
+//                    TC Wang 2022 - present.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          https://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/victimsnino/ReactivePlusPlus
+//
+
+#pragma once
+
+#include <rpp/observables/constraints.hpp>
+#include <rpp/operators/fwd/buffer.hpp>
+#include <rpp/subscribers/constraints.hpp>
+#include <rpp/sources/create.hpp>
+
+IMPLEMENTATION_FILE(buffer_tag);
+
+namespace rpp::details
+{
+
+/// A non-copyable class that provides a copyable on_next for the subscriber and
+/// allows copies of on_next(s) to share the same states.
+template<constraint::decayed_type UpstreamType, constraint::decayed_type DownstreamType>
+struct buffer_on_next : public std::enable_shared_from_this<buffer_on_next<UpstreamType, DownstreamType>>
+{
+    using DownstreamValueType = UpstreamType;
+
+    explicit buffer_on_next(const size_t& count) : max_(count) {};
```

Comment:
It is better to avoid references for PODs

```suggestion
    explicit buffer_on_next(size_t count) : max_(count) {};
```

Reply:
Do you have any recommended article giving more details?

Reply:
I found [this article](https://cplusplus.com/forum/beginner/261042/).

Reply:
One more

https://quuxplusone.github.io/blog/2021/11/09/pass-string-view-by-value/